analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -554,6 +556,35 @@ private String writeProjectNameIfNeeded() throws IOException {
         return newProjectName;
     }
 
+    private String extractProjectNameFromScmUrl(File rootDir) throws IOException {
+        File gitConfig = new File(new File(rootDir, ".git"), "config");
+        if (!gitConfig.isFile()) {
+            throw new ManipulationUncheckedException("No .git/config file found, failed to determine the root project name");
+        }
+        try {
+            List<String> lines = FileUtils.readLines(gitConfig, Charset.defaultCharset());
+            Optional<String> maybeScmUrlLine = lines.stream()
+                  .filter(line -> line.matches(SCM_URL_LINE_EXPR))
+                  .findFirst();
+
+            if (!maybeScmUrlLine.isPresent()) {
+                throw new ManipulationUncheckedException(
+                      ".git/config file doesn't define SCM URL, failed to determine the root project name");
+            }
+
+            // the line should be of form
+            // url = .../.../[^\\.]+(.git)?
```

Comment:
Although this looks correct to me, I suck at `regex`, so I'll let @rnc vouch for it

Reply:
I don't think this would match:
```
url = git@github.com:rnc/pom-manipulation-ext.git 
url = git+ssh://code.stage.foo.com/apache/kafka 
url = http://code.foo.com/gerrit/hibernate/hibernate-commons-annotations.git 
```

Could we use a Matcher and something like `\s*url\s*=(?:.*/)(.*)$` (although that still needs any ending of .git stripped off)

Reply:
Or maybe `\s*url\s*=(?:.*/)([a-zA-Z0-9\-]+)` ?